File: solutions/c/leap/1/leap.c
#include "leap.h"

bool divisible_by(int a, int b)
{
    return a % b == 0;
}

bool leap_year(int year)
{
    return divisible_by(year, 4) && !(divisible_by(year, 100) && !divisible_by(year, 400));
}

File: solutions/go/leap/1/leap.go
package leap

func IsLeapYear(year int) bool {
	is_year_divisible_by := func(n int) bool { return year%n == 0 }
	return is_year_divisible_by(4) &&
		!(is_year_divisible_by(100) && !is_year_divisible_by(400))
}
